client/app/bundles/course/duplication/components/TypeBadge/index.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { duplicableItemTypes } from 'course/duplication/constants';
+
+const styles = {
+  badge: {
+    padding: '2px 5px',
+    marginRight: 10,
+    borderStyle: 'solid',
+    borderRadius: 5,
+    borderWidth: 1,
+    fontSize: 12,
+  },
+};
+
+const translations = defineMessages({
+  [duplicableItemTypes.ASSESSMENT]: {
+    id: 'course.duplication.TypeBadge.assessment',
+    defaultMessage: 'Assessment',
+  },
+  [duplicableItemTypes.CATEGORY]: {
+    id: 'course.duplication.TypeBadge.category',
+    defaultMessage: 'Category',
+  },
+  [duplicableItemTypes.TAB]: {
+    id: 'course.duplication.TypeBadge.tab',
+    defaultMessage: 'Tab',
+  },
+});
+
+const TypeBadge = ({ text, itemType }) => (
```

Comment:
What do you think if we extend this `TypeBadge` to include the `Checkbox` element? There is some implicit requirement on how the duplication params are structured, and are repeated across the elements. Not sure if you met with any issues trying to do that.

Reply:
Sorry, don't really get what you mean for the second part. Can give an example of one of the implicit requirements you are referring to?

I think we should keep it as a presentational component. Instead, we can abstract out another component that includes both the `TypeBadge` and `Checkbox`, plus the indenting logic. I intend to do it after the survey cherry-pick PR - I'm gonna add more complexity and am not sure of the requirements yet, e.g. whether to embed features like "de/select all" or tree-collapsing, which might determine whether we need a `CheckboxTree` component or just a `CheckboxRow` component.

Reply:
> Sorry, don't really get what you mean for the second part. Can give an example of one of the implicit requirements you are referring to?

The duplication params have a certain convention - eg. `{ 'TAB' => [2,3,4] } `

> I think we should keep it as a presentational component. Instead, we can abstract out another component that includes both the `TypeBadge` and `Checkbox`, plus the indenting logic. I intend to do it after the survey cherry-pick PR - I'm gonna add more complexity and am not sure of the requirements yet, e.g. whether to embed features like "de/select all" or tree-collapsing, which might determine whether we need a `CheckboxTree` component or just a `CheckboxRow` component.

Ok, then we KIV this, and I'll merge it first. 👍